PennMobile/HomeViewController.swift

```diff
@@ -0,0 +1,312 @@
+//
+//  HomeViewController.swift
+//  PennMobile
+//
+//  Created by Josh Doman on 3/4/17.
+//  Copyright © 2016 PennLabs. All rights reserved.
+//
+
+import UIKit
+
+@objc class HomeViewController: GenericTableViewController {
+    
+    //cell identifiers
+    internal let weatherCell = "Weather"
+    internal let agendaCell = "Schedule"
+    internal let reservationCell = "Study Room Booking"
+    internal let diningCell = "Dining"
+    
+    //The cells or widgets that appear in Home Controller in order that they appear
+    lazy var customSettings: [String] = {
+        return [self.weatherCell, self.agendaCell, self.reservationCell, self.diningCell]
+    }()
+    
+    //Dining Halls for Dining Cell
+    var diningHalls: [DiningHall]!
+    
+    //Events for Agenda Cell
+    var events: [Event] = {
+        let event1 = Event(name: "LGST101", location: "SHDH 211", startTime: Time(hour: 8, minutes: 0, isAm: true), endTime: Time(hour: 9, minutes: 0, isAm: true))
+        
+        let event2 = Event(name: "MEAM101", location: "TOWN 101", startTime: Time(hour: 9, minutes: 0, isAm: true), endTime: Time(hour: 11, minutes: 0, isAm: true))
+        
+        let event3 = Event(name: "FNAR264", location: "FSHR 203", startTime: Time(hour: 11, minutes: 0, isAm: true), endTime: Time(hour: 12, minutes: 0, isAm: false))
+        
+        let event4 = Event(name: "MATH240", location: "HUNT 250", startTime: Time(hour: 11, minutes: 0, isAm: true), endTime: Time(hour: 3, minutes: 0, isAm: false))
+        
+        let event5 = Event(name: "GSWS101", location: "WILL 027", startTime: Time(hour: 1, minutes: 0, isAm: false), endTime: Time(hour: 2, minutes: 0, isAm: false))
+        
+        let event6 = Event(name: "CIS160", location: "MOOR 100", startTime: Time(hour: 7, minutes: 0, isAm: true), endTime: Time(hour: 2, minutes: 0, isAm: false))
+        
+        let event7 = Event(name: "PennQuest", location: "Houston Hall", startTime: Time(hour: 8, minutes: 30, isAm: true), endTime: Time(hour: 10, minutes: 30, isAm: true))
+        
+        return [event1, event2, event3, event4, event5, event6, event7]
+    }()
```

Comment:
Please remove these test cases and instead use the default "nothing" beard man.  Also, might be better to completely remove this from home page until we have an api for this.